packages/wonder-blocks-dropdown/components/dropdown.test.js

```diff
@@ -0,0 +1,379 @@
+                        populatedProps: {},
+                    },
+                ]}
+                keyboard={true}
+                light={false}
+                open={false}
+                // mock the opener elements
+                opener={dummyOpener}
+                openerElement={null}
+                onOpenChanged={openChanged}
+            />,
+        );
+    });
+
+    afterEach(() => {
+        unmountAll();
+    });
+
+    it("handles basic keyboard navigation as expected", () => {
+        const handleOpen = jest.fn();
+        dropdown.setProps({
+            initialFocusedIndex: 0,
+            keyboard: true,
+            onOpenChanged: (open) => handleOpen(open),
+            open: true,
+        });
+
+        const option0 = dropdown.find("OptionItem").at(0);
+        const option1 = dropdown.find("OptionItem").at(1);
+        const option2 = dropdown.find("OptionItem").at(2);
+
+        expect(dropdown.instance().focusedIndex).toBe(0);
+
+        // navigate down three times
+        option0.simulate("keydown", {keyCode: keyCodes.down});
+        option0.simulate("keyup", {keyCode: keyCodes.down});
+        expect(dropdown.instance().focusedIndex).toBe(1);
+
+        option1.simulate("keydown", {keyCode: keyCodes.down});
+        option1.simulate("keyup", {keyCode: keyCodes.down});
+        expect(dropdown.instance().focusedIndex).toBe(2);
+
+        option2.simulate("keydown", {keyCode: keyCodes.down});
+        option2.simulate("keyup", {keyCode: keyCodes.down});
+        expect(dropdown.instance().focusedIndex).toBe(0);
+
+        // navigate up back to last item
+        option0.simulate("keydown", {keyCode: keyCodes.up});
+        option0.simulate("keyup", {keyCode: keyCodes.up});
+        expect(dropdown.instance().focusedIndex).toBe(2);
+    });
+
+    it("closes on tab and escape as expected", () => {
+        const handleOpen = jest.fn();
+        dropdown.setProps({
+            onOpenChanged: (open) => handleOpen(open),
+            open: true,
+        });
+
+        const option0 = dropdown.find("OptionItem").at(0);
+
+        // "close" some menus
+        option0.simulate("keydown", {keyCode: keyCodes.tab});
+        option0.simulate("keyup", {keyCode: keyCodes.tab});
+        option0.simulate("keydown", {keyCode: keyCodes.escape});
+        option0.simulate("keyup", {keyCode: keyCodes.escape});
+        expect(handleOpen).toHaveBeenCalledTimes(2);
+        expect(handleOpen.mock.calls[0][0]).toBe(false);
+        expect(handleOpen.mock.calls[1][0]).toBe(false);
+    });
+
+    it("closes on external mouse click", () => {
+        const handleOpen = jest.fn();
+        dropdown.setProps({
+            onOpenChanged: (open) => handleOpen(open),
+            open: true,
+        });
+
+        const event = new MouseEvent("mouseup");
+        document.dispatchEvent(event);
+
+        expect(handleOpen).toHaveBeenCalledTimes(1);
+        expect(handleOpen.mock.calls[0][0]).toBe(false);
```

Comment:
This is b/c we're passing `false` to `handleOpen` b/c it's closed now.

Reply:
Added a comment to clarify!